### PR TITLE
add postgresql12 for rhscl and centos7

### PR DESCRIPTION
**Pull request by pkubatrh**
Some changes needed from v10:
replication configuration has been moved from recovery.conf to postgresql.conf
standby.signal file is now used to switch into non-primary mode

### CODE REVIEW

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
The openshift failure on rhel7 is expected as the image is not released yet

**Comment by pkubatrh** [comment]
[test-openshift]

File: src/root/usr/share/container-scripts/postgresql/openshift-custom-recovery.conf.template [hunk 1]
```diff
@@ -5,5 +5,7 @@
 #       Changes to this file will be overwritten.
 #
 
+{% if spec.version in ["9.4", "9.5", "9.6", "10"] %}
```
Comment (phracek) [inline src/root/usr/share/container-scripts/postgresql/openshift-custom-recovery.conf.template]:
```suggestion
{% if spec.version in ["9.6", "10", "12"] %}
```

File: src/root/usr/bin/run-postgresql-slave [hunk 2]
```diff
@@ -17,11 +17,19 @@ function initialize_replica() {
...
+{% if spec.version not in ["9.4", "9.5", "9.6", "10"] %}
+  cat >> "$PGDATA/postgresql.auto.conf" <<EOF
+{% else %}
   cat >> "$PGDATA/recovery.conf" <<EOF
+{% endif %}
 
 # Custom OpenShift recovery configuration:
 include '${POSTGRESQL_RECOVERY_FILE}'
 EOF
+{% if spec.version not in ["9.4", "9.5", "9.6", "10"] %}
```
Comment (phracek) [inline src/root/usr/bin/run-postgresql-slave]:
```suggestion
{% if spec.version not in ["9.6", "10"] %}
```

File: src/root/usr/bin/run-postgresql-slave [hunk 3]
```diff
@@ -17,11 +17,19 @@ function initialize_replica() {
 
   # PostgreSQL recovery configuration.
   generate_postgresql_recovery_config
+{% if spec.version not in ["9.4", "9.5", "9.6", "10"] %}
```
Comment (phracek) [inline src/root/usr/bin/run-postgresql-slave]:
```suggestion
{% if spec.version not in ["9.6", "10"] %}
```

File: test/run_test [hunk 4]
```diff
@@ -640,7 +641,7 @@ run_upgrade_test ()
     # is fixed for non-SCL use cases
     { [ "${OS}" == "fedora" ] || [ "${OS}" == "rhel8" ]; } && return 0
 
-    local upgrade_path="none 9.2 9.4 9.5 9.6 10 none" prev= act=
+    local upgrade_path="none 9.2 9.4 9.5 9.6 10 12 none" prev= act=
```
[the remark this answers is not in the document]
Comment (pkubatrh) [reply test/run_test]: We need to keep this, this part tests the upgrade path of different versions of the images. While we do not support the images anymore we should support upgrading to a new version from them.

File: test/run_test [hunk 5]
```diff
@@ -659,9 +660,9 @@ run_migration_test ()
 
     local from_version
     if [ "${OS}" == "rhel8" ]; then
-        local upgrade_path="9.6 10"
+        local upgrade_path="9.6 10 12"
     else
-        local upgrade_path="9.2 9.4 9.5 9.6 10"
+        local upgrade_path="9.2 9.4 9.5 9.6 10 12"
```
[the remark this answers is not in the document]
Comment (pkubatrh) [reply test/run_test]: Same here

**Comment by pkubatrh** [comment]
[test-openshift]

File: specs/multispec.yml [hunk 6]
```diff
@@ -91,9 +91,28 @@ specs:
...
+      has_devel_repo:
+        rhel: 'brew'
+        centos: True
+
 matrix:
   exclude:
     - distros:
         - fedora-29-x86_64
         - rhel-8-x86_64
       version: "9.5"
```
[the remark this answers is not in the document]
Comment (pkubatrh) [reply specs/multispec.yml]: This is out of scope for this PR.

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test][test-openshift]

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
@phracek PTAL

The fail in centos openshift CI is expected as the image is not yet on dockerhub. It is missing something similar to https://github.com/sclorg/postgresql-container/pull/343/commits/ab972e230edf4208e14b13f95623980521444418 but I am thinking of moving the check function to common scripts instead as we have it in multiple places already. Will open a follow-up after it is available.

**Comment by phracek** [comment]
Thanks for the changes!. Awesome work.

**Comment by pkubatrh** [comment]
Thanks for the review! Merging

**Pull request merged**